### PR TITLE
chore: export types & logics from widget

### PR DESCRIPTION
**Pull request by samobasquiat**
# Summary

There are some dependencies to @rango-dev/... (helpers, types, ...) in dapp and should be removed, so we should export them from embedded.

Fixes # (issue)
There is another PR here to test: 
https://github.com/rango-exchange/app/pull/14

# How did you test this change?

# Checklist:

- [x] I have performed a self-review of my code

### CODE REVIEW

File: widget/embedded/src/index.ts [hunk 1]
```diff
@@ -84,9 +106,14 @@ export {
   useWallets,
   useWidgetEvents,
   customizedThemeTokens,
+  readAccountAddress,
   MainEvents,
+  Networks,
```
Comment (yeager-eren) [inline widget/embedded/src/index.ts:111]: I'm a bit concerned about exporting so many things from `embedded`. Some of them is Rango specific I believe and will be deprecated soon like `Networks` and `WalletTypes`.

For more explicitly separate them, I suggest to export them using a separate `export`.

```
export {
  useWallets,
  useWidgetEvents,
  customizedThemeTokens,
  ...
 }
 
 // Internal exports for Rango
 export {
 Networks,
 WalletTypes,
 ...
 }
```

 We can do same thing for `exprot type { ...}` as well.
 
 The final goal keeping these internal things separated somehow, only put them on main api (export) when we are sure they should be there.

File: wallets/react/src/types.ts [hunk 2]
```diff
@@ -20,8 +21,6 @@ export type ConnectResult = {
   provider: any;
 };
 
-export type Providers = { [type in WalletType]?: any };
```
Comment (yeager-eren) [inline wallets/react/src/types.ts:23]: It's part of the `wallets/react` API and should be kept here. Please revert this change.

Comment (samobasquiat) [reply wallets/react/src/types.ts]: fixed.
